### PR TITLE
bitbake/yocto_recipe.py: added dynamic license year, and functionality within get_recipe_text for if the license is a list instead of a string    CLOSING TO REBASE

### PR DESCRIPTION
**Pull request by shaneallcroft**

### CODE REVIEW

**Comment by shaneallcroft** [comment]
Oh God all my changes from my last request back to haunt me

**Comment by shaneallcroft** [comment]
My commits arent very descriptive, so not that it matters a ton for this particular instance , but here were the issues solved in order: 

I had a semicolon at the end of setting first to False, and that i didn't capitalize the t in true

I used !first in the if statement when that's illegal

import time was after "from urllib.request import urlretrieve" when it should have been before

**Comment by shaneallcroft** [comment]
I totally forgot that Travis supplied multiple errors at a time what the heck I'm a huge dork

**Comment by shaneallcroft** [comment]
Oh my God, the commented out code was VERBATIM what was needed : (  Haha I completely forgot that Python does its booleans using english words, and I thought the comment was Pseudo code needed to be written. I'm a fungus.

File: superflore/generators/bitbake/yocto_recipe.py [hunk 1]
```diff
@@ -136,16 +138,13 @@ def get_recipe_text(self, distributor, license_text, die_msg=None):
         elif isinstance(self.license, list):
             self.license = self.license[0].replace(' ', '-')
             ret += 'LICENSE = "' + self.license + '"\n'
-            """
-            TODO(allenh1): add this functionality
```
Comment (tfoote) [inline superflore/generators/bitbake/yocto_recipe.py]: This functionality should use `' '.join(self.license)` to be more pythonic and compact.

Comment (shaneallcroft) [reply superflore/generators/bitbake/yocto_recipe.py]: Thank you for the feedback! I'll implement that, and check out rebasing rn :+1:

**Comment by shaneallcroft** [comment]
Nope guess I didnt rebase stand by

**Comment by shaneallcroft** [comment]
Wait i think I get it now

**Comment by allenh1** [comment]
Ok, this is coming out of your master branch, it looks like.

```
$ git remote add ros-infrastructure https://github.com/ros-infrastructure/superflore
$ git fetch --all
$ git checkout -b rebase origin/master
$ git rebase -i ros-infrastructure/master
```

---

I'm not sure what @tfoote would do at this point, but I might do something absolutely crazy like this.

```
$ git push -u origin rebase
$ git branch -D master
$ git checkout -b master
$ git log
```

MAKE SURE THE LOG IS WHAT YOU EXPECT. You cannot recover your work if you type the next line and you're wrong.

```
$ git push -u origin master --force
```

**Pull request closed without merging**